harness/determined/common/experimental/determined.py

```diff
@@ -404,3 +404,157 @@ def stream_trials_validation_metrics(
             trial_ids: List of trial IDs to get metrics for.
         """
         return trial._stream_validation_metrics(self._session, trial_ids)
+
+    def get_workspace_ids(
+        self,
+        session: api.Session,
+        workspace_names: List[str],
+    ) -> List[int]:
+        wnSet = set(workspace_names)
+        workspace_ids = []
+
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspacesResponse:
+            return bindings.get_GetWorkspaces(
+                session=self._session,
+                offset=offset,
+            )
+
+        resps = api.read_paginated(get_with_offset)
+        workspaces = [w for r in resps for w in r.workspaces]
+
+        for w in workspaces:
+            if w.name in wnSet:
+                if w.archived:
+                    raise Exception(f'Workspace "{w.name}" is archived.')
+                workspace_ids.append(w.id)
+        return workspace_ids
+
+    def bind_rps_to_workspaces(
+        self,
+        resource_pool_names: List[str],
```

Comment:
I believe we only want to bind one resource pool at a time, to keep it consistent with the API and webui

Reply:
Cool, that makes more sense. I will update the PRD.